sys-utils/libcmt/README.md

```diff
@@ -134,6 +134,79 @@ cast calldata-decode "Notice(bytes)" 0x`xxd -p -c0 "$1"` | (
 This library is intented to be used with programming languages other than C.
 They acheive this by different means.
 
+## Python
+
+Python has multiple Foreign Function Interface (FFI) options for interoperability with C.
+This example uses CFFI and the `libcmt` mock. Which is assumed to be installed at `./libcmt-0.1.0`.
+
+This document uses the [main mode](https://cffi.readthedocs.io/en/latest/overview.html#main-mode-of-usage) of CFFI and works in two steps: `build`, then `use`.
+
+### Build
+
+The `build` step has the objective of creating a python module for libcmt.
+To acheive this we'll use `libcmt/ffi.h` and the script below.
+Paths may need adjustments.
+
+```
+import os
+from cffi import FFI
+
+ffi = FFI()
+with open(os.path.join(os.path.dirname(__file__), "../libcmt-0.1.0/usr/include/libcmt/ffi.h")) as f:
+    ffi.cdef(f.read())
+ffi.set_source("pycmt",
+"""
+#include "libcmt/rollup.h"
+""",
+     include_dirs=["libcmt-0.1.0/usr/include"],
+     library_dirs=["libcmt-0.1.0/usr/lib"],
+     libraries=['cmt'])
+
+if __name__ == "__main__":
+    ffi.compile(verbose=True)
+```
+
+### Use
+
+With the module built, we can import it with python and use its functions.
+This example uses the raw bindings and just serves to ilustrate the process.
+Better yet would be to wrap these functions into a more "pythonic" API.
+
+`LD_LIBRARY_PATH=libcmt-0.1.0/lib/ python`
+
+```
+#!/usr/bin/env python
+
+# run this file only after building pycmt!
+import os
+from pycmt.lib import ffi, lib
+
+r = ffi.new("cmt_rollup_t[1]")
+assert(lib.cmt_rollup_init(r) == 0)
+
+address = ffi.new("uint8_t[20]", b"1000000000000")
+value   = ffi.new("uint8_t[32]", b"0000000000001")
```

Comment:
Isn't this leaking? We should provide a correct example without leaks.

Reply:
I don't think so, from docs:
```
Unlike C, the returned pointer object has ownership on the allocated memory: when this exact object is garbage-collected, then the memory is freed.
```
ref. https://cffi.readthedocs.io/en/latest/using.html#id16

Reply:
Ok, got it :). Python stuff.